### PR TITLE
udn: rename the annotation used

### PR DESCRIPTION
**Pull request by maiqueb**
**What this PR does / why we need it**:
This PR renames the annotation used to match the annotation expected [in this OVN-Kubernetes PR](https://github.com/ovn-org/ovn-kubernetes/pull/4651/).

In the future, we may actually use a label, and this name might change.

We expect to define this in this [OpenShift enhancement](https://github.com/openshift/enhancements/pull/1673).

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/config/types.go [hunk 1]
```diff
@@ -11,7 +11,7 @@ const (
 	NetworkRolePrimary NetworkRole = "primary"
 )
 
-const OVNPrimaryNetworkIPAMClaimAnnotation = "k8s.ovn.org/ovn-udn-ipamclaim-reference"
+const OVNPrimaryNetworkIPAMClaimAnnotation = "k8s.ovn.org/primary-udn-ipamclaim"
```
Comment (qinqon) [inline pkg/config/types.go:14]: We need also 
```
pkg/ipamclaimswebhook/podmutator_test.go:                                               Path:      "/metadata/annotations/k8s.ovn.org~1ovn-udn-ipamclaim-reference",
pkg/ipamclaimswebhook/podmutator_test.go:                                               Path:      "/metadata/annotations/k8s.ovn.org~1ovn-udn-ipamclaim-reference",

```

**Review by qinqon — CHANGES_REQUESTED** [review]
Fix tests

File: pkg/config/types.go (hunk 1 above)
Comment (maiqueb) [reply pkg/config/types.go:14]: Good catch !

**Pull request merged**